benchopt/cli/main.py

```diff
@@ -182,15 +188,21 @@ def run(config_file=None, **kwargs):
 
     (
         benchmark, solver_names, forced_solvers, dataset_names,
-        objective_filters, max_runs, n_repetitions, timeout, n_jobs, slurm,
+        objective_filters, max_runs, n_repetitions, timeout, no_timeout, n_jobs, slurm,
         collect, plot, display, html, pdb, do_profile, env_name, output_name
     ) = _get_run_args(kwargs, config)
 
-    try:
-        timeout = int(float(timeout))
-    except ValueError:  # already under string format
-        import pandas as pd
-        timeout = pd.to_timedelta(timeout).total_seconds()
+    # If --no-timeout is set and --timeout is not, skip this block and keep timeout=None
+    if timeout != None and no_timeout:
+        raise Exception('You cannot specify both --timeout and --no-timeout options.')
+    elif timeout == None and no_timeout == False:
```

Comment:
Usually, we don't use `== False` to check boolean conditions, simply use `no_timeout/not no_timeout`
```suggestion
    elif timeout is None and not no_timeout:
```